### DISCUSSION TITLE
Participating to translation

### DISCUSSION THREAD

**Discussion by skrilax91** [0xJacky/nginx-ui#114] [Q&A] [answered]
I wonder if there is documentation on how to help with the translation of the project?
I would like to propose a French translation but I don't know how to proceed

**Comment by Hintay** [comment, marked as answer]
Hi there! Thank you for your interest in helping with the translation of the project and for considering proposing a French translation.

I apologies for not having a document to help you start the translation process, I will add this section to the documentation later. First, the language files are located in the `frontend/src/language` directory. You will need to use a po file editor (I recommend using Poedit) to translate the language. 

Here are the steps you can follow:

1. Create a folder with the name of the language code in the `language` directory. In your case, it should be `fr` (or `fr_FR`, depend on your choice).
2. Generate your own `app.po` file for the language you want to translate by po file editor based on the translation template `messages.pot`, and place it in the `fr` folder.
3. Add the `fr` code to the `LINGUAS` file, and add the available language code in `frontend/src/gettext.ts`.
4. Finally, return to the `frontend` directory and run `yarn gettext:compile` to generate the required json files for the frontend.

If you are not familiar with programming, you can simply follow the first two steps. After completing these steps, you can propose a pull request for the project according to the [guide](https://nginxui.com/guide/contributing.html).

I hope this helps! Thank you for your contribution to the project.

**Reply by Hintay** [reply]
Also, we are considering using [weblate](https://weblate.org/) to simplify the translation process in the future.

**Comment by skrilax91** [comment]
Thanks for your reply !
